system/logging/appenders/FileAppender.cfc

```diff
@@ -306,8 +306,10 @@ component accessors="true" extends="coldbox.system.logging.AbstractAppender"{
 	 * @message The target message
 	 */
 	private FileAppender function append( required message ){
-		// If we are not in a thread, then start the log listener, else queue it
-		if( !getUtil().inThread() ){
+		var util = getUtil();
+
+		// if thread within thread not supported, check we're not already in child thread
+		if( util.threadWithinThreadSupported() || !util.inThread() ){
```

Comment:
Actually, out of curiosity, what would happen if we just don't do the check? Will the CFML engines create the `cfthread` or just thrown an exception?

Reply:
Lucee, prior to the fix, would throw an exception before creating thread (no thread created). 
Open BlueDragon would always allow it
ACF - ? I dunno. 

But yes, the thread tag could just be wrapped in a try catch. But that is perhaps less efficient (would need to perf test that for memory usage, etc)?